changelog/no-cmpsb.dd

```diff
@@ -0,0 +1,26 @@
+`memcmp()` compares are no longer performed for struct equality tests, memberwise comparisons are done instead, per the spec
+
+The compiler would sometimes generate code for struct equality tests using
+`memcmp()` across the whole struct object. This assumed that:
+
+$(OL
+$(LI alignment holes were filled with 0, which is not always the case)
+$(LI there were no floating point NaN values, which should compare as not equal
+even if the bit patterns match)
+)
+
+The spec requires that the comparison be done in a memberwise manner.
+This brings the implementation in line with this.
+
+$(LINK2 https://dlang.org/spec/expression.html#equality_expressions, Equality Expressions)
```

Comment:
This links stands a bit out in the flow of the reader:

![image](https://user-images.githubusercontent.com/4370550/52514367-47ab8500-2c11-11e9-98c6-29bf6ce41ce9.png)


Maybe directly link it as e,g,  "[Equality Expressions](...) can break ..."?